lesson16/index.js

```diff
@@ -0,0 +1,158 @@
+// const url = "https://myjson.dit.upm.es/api/bins/3aux";
+const url = "https://myjson.dit.upm.es/api/bins/hmd9";
+let allData = {};
+const newsBlock = document.querySelector(".news");
+const newsTabs = document.querySelector(".news_Tabs");
+
+const createNewsTabs = (allData) => {
+  allData.forEach((content, index) => {
+    const li = document.createElement("li");
+    li.classList.add("tab_item");
+    li.textContent = content.category;
+    newsTabs.appendChild(li);
+  });
+};
+
+const createNewsContent = (allData) => {
+  // 初期表示用の処理
+  const primaryElement = document.createElement("div");
+  primaryElement.classList.add("news_Wrapper");
+  const lists = document.createElement("ul");
+  lists.classList.add("news_Lists");
+  const primaryImageWrapper = document.createElement("div");
+  primaryElement.appendChild(lists);
+  primaryElement.appendChild(primaryImageWrapper);
+  newsBlock.appendChild(primaryElement);
+  allData.forEach((content, index) => {
+    if (content.isActive) {
+      primaryImageWrapper.classList.add("news_Img");
+      const img = document.createElement("img");
+      img.src = content.image.img;
+      primaryImageWrapper.appendChild(img);
+
+      renderNewsList(content, lists);
+      const tabItem = document.querySelector(
+        ".news_Tabs .tab_item:nth-child(1)"
+      );
+      tabItem.classList.add("is-active");
+    }
+  });
+
+  // タブクリック時の処理
+  newsTabs.addEventListener("click", (event) => {
+    // event.targetにis-activeをつける
+    // event.targetのidを取得し、それに紐づく記事と画像を取得
+    const target = event.target;
+    const tabsItems = Array.from(newsTabs.querySelectorAll("li"));
+
+    // タブのindexを取得し、スタイル変更
+    tabsItems.forEach((tab) => {
+      tab.addEventListener("click", (e) => {
+        const activeIndex = tabsItems.filter((tab) => {
+          return tab === e.target;
+        });
+        const inactiveIndex = tabsItems.filter((tab) => {
+          return tab !== e.target;
+        });
+
+        activeIndex[0].classList.add("is-active");
+        inactiveIndex.forEach((element) => {
+          element.classList.remove("is-active");
+        });
+      });
+    });
```

Comment:
IMO
You can also write something like this
```suggestion
  const activeTab = newsTabs.querySelector(".is-active");
    activeTab.classList.remove("is-active");
    target.classList.add('is-active');
```

Reply:
I thought it was a very efficient method. Thank you very much!